includes/blocks.inc

```diff
@@ -684,16 +690,41 @@ function _islandora_solr_advanced_search_terms($form, &$form_state) {
  */
 function islandora_solr_advanced_search_form_submit($form, &$form_state) {
   module_load_include('inc', 'islandora_solr', 'includes/utilities');
+
   // Collect query values.
   $query_array = array();
+
+  // Get Lucene syntax escaping configuration, prior to the following
+  // foreach loop.
+  $lucene_syntax_escape = variable_get('islandora_solr_advanced_search_block_lucene_syntax_escape', FALSE);
+  $lucene_regex_default = variable_get(
+    'islandora_solr_advanced_search_block_lucene_regex_default',
+    ISLANDORA_SOLR_QUERY_FACET_LUCENE_ESCAPE_REGEX
+  );
   foreach ($form_state['values']['terms'] as $term) {
     $field = $term['field'];
     $search = trim($term['search']);
-    $search = islandora_solr_replace_slashes($search);
+
+    // Only replace slashes here (~slsh~) when automatic Lucene Syntax Escaping
+    // is disabled as the strings they are replace with (~slsh~ and ~plus~)
+    // contain the character '~' that would be escaped when automatic Lucene
+    // Syntax Escaping is enabled.
+    if (!$lucene_syntax_escape) {
+      $search = islandora_solr_replace_slashes($search);
+    }
+
     $boolean = (isset($term['boolean'])) ? $term['boolean'] : variable_get('islandora_solr_search_boolean', 'user');
     $field = islandora_solr_lesser_escape($field);
     // Add query.
     if (!empty($search)) {
+      if ($lucene_syntax_escape) {
+        $search = islandora_solr_facet_query_escape($search, $lucene_regex_default);
+
+        // Replace the occurances of '~slsh~' and '~plus~' after
+        // islandora_solr_facet_query_escape(), since they both contain
+        // '~', which would be improperlly escaped prior to query execution.
+        $search = islandora_solr_replace_slashes($search);
```

Comment:
Instead of hitting `islandora_solr_replace_slashes()` two places as it does now, why not move it down to immediately before adding to the `$query_array` below?... Could then have just the one call...

... Really should almost go lower, when defining the `$form_state['redirect']`, on all of `$query`, so we hit any funky "base" queries which might be defined.